src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncIteratorMethodToStateMachineRewriter.cs

```diff
@@ -90,6 +91,8 @@ protected override BoundStatement GenerateSetResultCall()
             AddDisposeCombinedTokensIfNeeded(builder);
 
             builder.AddRange(
+                // _current = default;
+                GenerateClearCurrent(),
```

Comment:
>GenerateClearCurrent() [](http://example.com/codeflow?start=16&length=22)

It looks like the order of evaluation doesn't match the comment above. Which order is intended?

Reply:
I don't follow. The new statement is added after the disposal of `combinedTokens` and before calling method on `promiseOfValueOrEnd`. I think OP description matches the implementation.

Reply:
Oh, I see. The comment above doesn't match order here. I'll correct. Thanks